portability-core/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudModule.java

```diff
@@ -59,4 +75,36 @@ GoogleCredentials getCredentials(CommonSettings commonSettings) throws GoogleCre
     }
   }
 
+  /**
+   * Get project ID from environment variable and validate it is set.
+   *
+   * @throws IllegalArgumentException if project ID is unset
+   */
+  @Provides @Singleton ProjectId getProjectId(CommonSettings commonSettings) {
```

Comment:
Could we use Guice bind(String.class).annotatedWith(ProjectId.class).to("the env var") to avoid having a wrapper class?

Reply:
I just tried @ProjectId String projectId again with the old binding and it worked. Not sure what I was doing before that caused eager eval. Thanks for bringing it up, it is much better without the wrapper class!